profiles/lib/tag_cache.lua

```diff
@@ -0,0 +1,35 @@
+TagCache = {}
+
+function TagCache.get(data, key)
+  local v = data.cache[key]
+  if v then
+    if v == '' then
+      return nil
+    else
+      return v
+    end
+  else
+    v = data.way:get_value_by_key(key)
+    if v == nil then
+      data.cache[key] = ''
+      return nil
+    else
+      data.cache[key] = v
+      if v == '' then
+        return nil
+      else
+        return v
+      end
+    end
+  end
+end
+
+function TagCache.set(data,key,value)
```

Comment:
the existing profile sometimes modifies the locally stored tag value, this functions support that mechanism